### PR TITLE
Introducing mizu logs dump & Log prints alignment in API server using rlog

### PR DESCRIPTION
**Pull request by IgorGov**

### CODE REVIEW

**Review by gadotroee — COMMENTED** [review]
looking pretty simple and good

**Review by gadotroee — DISMISSED** [review]

File: cli/kubernetes/provider.go [hunk 1]
```diff
@@ -466,6 +473,19 @@ func (provider *Provider) CheckPodExists(ctx context.Context, namespace string,
 	return false, nil
 }
 
+func (provider *Provider) GetPods(ctx context.Context, namespace string) ([]shared.PodInfo, error) {
```
Comment (nimrod-up9) [inline cli/kubernetes/provider.go:564]: Rename to ListPods in order to match k8s terms.

File: cli/kubernetes/provider.go [hunk 2]
```diff
@@ -52,7 +55,12 @@ func NewProvider(kubeConfigPath string) (*Provider, error) {
 	kubernetesConfig := loadKubernetesConfiguration(kubeConfigPath)
 	restClientConfig, err := kubernetesConfig.ClientConfig()
 	if err != nil {
-		return nil, err
+		if clientcmd.IsEmptyConfig(err) {
+			return nil, fmt.Errorf("Couldn't find the kube config file, or file is empty. Try adding '--kube-config=<path to kube config file>'\n")
+		}
+		if clientcmd.IsConfigurationInvalid(err) {
+			return nil, fmt.Errorf("Invalid kube config file. Try using a different config with '--kube-config=<path to kube config file>'\n")
+		}
```
Comment (nimrod-up9) [inline cli/kubernetes/provider.go:63]: Could be moved to `errormessage`.

File: cli/cmd/logs.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		ctx, _ := context.WithCancel(context.Background())
+
+		if filePath == "" {
+			pwd, err := os.Getwd()
+			if err != nil {
+				mizu.Log.Errorf("Failed to get PWD, %v (try using `mizu logs -f <full path dest zip file>)`", err)
+				return nil
+			}
+			filePath = path.Join(pwd, "mizu_logs.zip")
+		}
```
Comment (nimrod-up9) [inline cli/cmd/logs.go:32]: I thought you wanted these kind of operations to be a part of the config?

File: cli/logsUtils/mizuLogsUtils.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	if err := addFileToZip(zipWriter, mizu.GetLogFilePath()); err != nil {
+		mizu.Log.Errorf("Failed write file, %v", err)
+	} else {
+		mizu.Log.Infof("Successfully added file %s", mizu.GetLogFilePath())
+	}
+	mizu.Log.Infof("You can find the zip with all logs in %s\n", filePath)
+	return nil
+}
+
+func addFileToZip(zipWriter *zip.Writer, filename string) error {
```
Comment (nimrod-up9) [inline cli/logsUtils/mizuLogsUtils.go:60]: Is it worth to move this to zip utils?

File: cli/kubernetes/provider.go (hunk 2 above)
Comment (IgorGov) [reply cli/kubernetes/provider.go:63]: like

**Review by nimrod-up9 — APPROVED** [review]

**Pull request merged**